### PR TITLE
GO-2143 Manifest downloading function

### PR DESCRIPTION
**Pull request by KirillSto**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

This PR introduces RPC command for downloading Experience manifests

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2143/download-manifest-function

### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by fat-fellow — APPROVED** [review]

File: core/gallery/gallery.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+	return info, nil
+}
+
+func isInWhitelist(url string) bool {
+	if len(whitelist) == 0 {
+		return true
+	}
+	for _, wlElement := range whitelist {
+		if strings.Contains(url, wlElement) {
```
Comment (deff7) [inline core/gallery/gallery.go:98]: I can add `?waitlistbypass=anytype.io` to bypass whitelist

Comment (deff7) [reply core/gallery/gallery.go:98]: check url host

Comment (KirillSto) [reply core/gallery/gallery.go]: Modified whitelist var to map[string]Regexp, so we can analyze hosts and paths separately

File: core/gallery/gallery.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+const timeout = time.Second * 30
+
+type schemaResponse struct {
+	Schema string `json:"$schema"`
+}
+
+// whitelist maps allowed hosts to regular expressions of URL paths
+var whitelist = map[string]*regexp.Regexp{
+	"raw.githubusercontent.com": regexp.MustCompile(`/anyproto.*`),
```
Comment (deff7) [inline core/gallery/gallery.go:29]: this matches regexp as well:
```
/foo/bar/anyproto
```

maybe we need to add `^` at the beginning of regex?

**Pull request merged**